### DISCUSSION TITLE
Uno class library. Difference of versions.

### DISCUSSION THREAD

**Discussion by limeniye** [unoplatform/uno#13065] [General]
<h2>Steps</h2>

 - Create "Uno Platform App" project;
 - Add to created folder a new "Uno Platform Library" project.
 
 Expectations: All should be without warnings and error.
 Result: Some versions difference problem.

https://github.com/unoplatform/uno/assets/69314237/e6af170c-3703-4413-8ad6-7861d9f01ae4

<h2>Steps to fix this problem</h2>

 - Remove Uno.WinUI package version;
 - Remove Microsoft.WindowsAppSDK and Microsoft.Windows.SDK.BuildTools package versions.
 
https://github.com/unoplatform/uno/assets/69314237/20f0cad5-cd14-4923-a987-277366fd3325

**Comment by jeromelaban** [comment]
@nickrandolph

**Reply by limeniye** [reply]
Still actual

The library project are using the following old packages which has a conflicts:
```XML
"Microsoft.WindowsAppSDK" Version="1.2.230313.1"
"Uno.WinUI" Version="4.10.9"
```

**Comment by limeniye** [comment]
@jeromelaban, I see that it was fixed!
![image](https://github.com/unoplatform/uno/assets/69314237/a1a1de2f-96c1-4efa-9332-e58f67381b25)

But, as you can see the target .net version is staticly 7.0 and Microsoft.WindowsAppSDK is **1.3.230602002** but in Uno Application I have the .net 8.0 versions and **1.4.231008000** version of Microsoft.WindowsAppSDK.

I guess this problem will appear again in the future.
Does the Uno team have any plans for a global solution to this problem?
For example, creating another wizard, for a cross-platform library.
Thanks.

**Reply by jeromelaban** [reply]
Indeed. There is a conflict with the library not using CPM. This is a problem at this time, where the parameters are not provided by the template itself. 

@dansiegel we'll likely need to move the class library template in the uno.templates repo, and account for that so we make CPM possible and/or optional (if we don't move to CPM being mandatory).

**Reply by limeniye** [reply]
related reply: https://github.com/unoplatform/uno/discussions/14515#discussioncomment-7656965

**Comment by MarkBrainfire** [comment]
I'm glad to see this being discussed and addressed.  Yes, we need an Uno Library Wizard.

**Reply by ToddGlodek** [reply]
I agree

**Reply by limeniye** [reply]
I think you can leave your emoji here: https://github.com/unoplatform/uno.templates/issues/503
This will raise the issue to the top.
